### ISSUE TITLE
Video playback crashes, AmdD3D11CreateDeviceAndSwapChainExt access violation

### ISSUE THREAD

**Issue by CastagnaIT** [xbmc/xbmc#25985]
### Describe the bug
Here is a clear and concise description of what the problem is:

i recently changed video card from NVIDIA 1030 to AMD RX 5500 with clean install drivers
drivers used are the latest adrenaline 24.10.1

i have more videos that make kodi crashes when played usually after some seconds
im on Windows 11

Example of a video that make kodi crash
[1023_20150107033106.zip](https://github.com/user-attachments/files/17799074/1023_20150107033106.zip)

all crashes point to same error

### To Reproduce

1. play the video, after 3-4 secs crash
2.
3.

### Debuglog

The debuglog can be found here:
[kodi.log](https://github.com/user-attachments/files/17799066/kodi.log)
[kodi_crashlog-22.0-ALPHA1 Git_20241118-350be7bebd-dirty-20241118-113213.dmp](https://github.com/user-attachments/files/17799067/kodi_crashlog-22.0-ALPHA1.Git_20241118-350be7bebd-dirty-20241118-113213.dmp)
Textual dmp text: [DbgTxt.txt](https://github.com/user-attachments/files/17799103/DbgTxt.txt)

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] macOS
 - [x] Windows
 - [ ] Windows UWP

 - Operating system version/name: windows 11
 - Kodi version: master branch and nightly KodiSetup-20241115-65703fe9-master-x64.exe

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by thexai** [comment]
The sample is an interlaced video. Maybe is a side effect of https://github.com/xbmc/xbmc/pull/23606

@CrystalP

**Comment by CrystalP** [comment]
Yes it's most likely the AMD deinterlacing bug. The attached sample is interlaced.

The issue has been present at least in the last few major releases of Kodi and was not caused by that commit.
The forum thread is valid and has good information, all signs point to something AMD changed in the drivers of RDNA gpus after a certain version number (see forum thread).
It doesn't mean that the problem is entirely in the driver and that Kodi does everything right, but it didn't turn into a problem until that driver version.
I tried many changes and looked for other free software doing something similar. Most don't even engage the advanced deinterlacing algorithms (don't provide the future/past frames) and for the one that does I wasn't able to figure out the difference.

What's new is that somehow your debugger was able to get some symbols for the AMD driver (the AmdD3D11CreateDeviceAndSwapChainExt), but the rest of the stack isn't very helpful. It's a driver internal thread, and maybe something is causing a device or swap chain to be recreated, but it's not a direct result of a call by Kodi, more like an internal state change.

Next, ideally we'd find a way to get in touch with the AMD driver team to troubleshoot this.

**Comment by thexai** [comment]
> No problem with Nexus, but with Omega Kodi freezes

https://forum.kodi.tv/showthread.php?tid=372850&pid=3213833#pid3213833

Has it been verified if the same video sample with the same driver does not crash on Nexus?

**Comment by CastagnaIT** [comment]
i tested on kodi 20 but crashes
[kodi_crashlog-20.1 Git_20230312-289ec664e3-20241119-170230.dmp](https://github.com/user-attachments/files/17817981/kodi_crashlog-20.1.Git_20230312-289ec664e3-20241119-170230.dmp)
